src/components/forms/DatePicker/DatePicker.stories.tsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react'
+
+import { DatePicker } from './DatePicker'
+import { Form } from '../Form/Form'
+import { FormGroup } from '../FormGroup/FormGroup'
+import { Label } from '../Label/Label'
+
+export default {
+  title: 'Components/Form controls/Date picker',
+  component: DatePicker,
+  argTypes: {
+    onSubmit: { action: 'submitted' },
+    disabled: { control: { type: 'boolean' } },
+  },
+  parameters: {
+    docs: {
+      description: {
+        component: `
+### USWDS 2.0 DatePicker component
+Source: https://designsystem.digital.gov/components/form-controls/#DatePicker
+
+**Note:** There is one small difference in functionality between this component and the USWDS implementation, related to validating the input. The USWDS implementation validates when:
+- setting the initial value based on the default value passed in
+- clicking on a date in the calendar UI
+- typing the Enter key in the external text input
+- on focusout (blur) of the external text input
+
+Because this component uses the useEffect hook to trigger validation whenever the date value changes (regardless of how), the React DatePicker will validate when:
+- setting the initial value based on the default value passed in (same as above)
+- clicking on a date in the calendar UI (same as above)
+- on input (change) of the external text input
+
+It's also worth mentioning that validation in this case is just calling [setCustomValidity](https://developer.mozilla.org/en-US/docs/Web/API/HTMLObjectElement/setCustomValidity) on the external text input, and library users should be able to determine how & when they want invalid UI to display by inspecting the [ValidityState](https://developer.mozilla.org/en-US/docs/Web/API/ValidityState) of the external input.
```

Comment:
Is it worth adding a storybook story demonstrating this? I was able to test the validity as follows:
1. Assign an id to the `input` in a storybook story
2. Enter an invalid date
3. Check validity via `document.getElementById('my-id').checkValidity()`
    - returns `false`
4. Select/enter valid date
    - returns `true`

I wouldn't hold up this PR just for this story, so maybe we make it an issue as a nice-to-have for the storybook docs?

Reply:
after thinking about this I started going down a bit of a rabbit hole regarding exposing refs to input components that are more complex and use refs to the inputs internally themselves. I believe I have somewhat figured it out, but it’s a new chunk of code that to me feels like maybe outside of “mvp” for DatePicker. the factors I’m considering are:
- exposing refs to input components is usually important so that consumers can use them to do things like imperatively focus on an input, and check the native validity state of it
- I think the answer is to use `React.forwardRef` on the DatePicker component, and `useImperativeHandle` to expose important methods/properties back to the parent ref
- this was mostly through exploration from the last hour or so, and I don’t feel comfortable merging these additions without more known use cases of where this feature is needed
- React very explicitly says if you start using `forwardRef` in a library, treat it as a breaking change. Meaning if we merge the DatePicker without it, and then we add it in, it will be a major version update
- I have a suspicion that the ComboBox component will need similar treatment, so adding `forwardRef` there would also require a major version update

based on all of the above, this is my inclination:
- Merge the DatePicker component as-is, with no exposed ref
- Do some more exploratory work, using the Example app or other consumer projects, to use the DatePicker and ComboBox inputs in Real Life Forms and feel out the limitations caused by not having access to the input ref
- Create a new issue for resolving the above, with the assumption that this will most likely require using `fowardRef` on these components and trigger a new major version of the library

References:
- https://reactjs.org/docs/forwarding-refs.html#note-for-component-library-maintainers
- https://reactjs.org/docs/hooks-reference.html#useimperativehandle
- https://medium.com/@ilxanlar/useimperativehandle-by-examples-99cbdc8e3c3a